### PR TITLE
Checkbox button widget

### PR DESCRIPTION
**Pull request by robinr**
Implementation of Checkbox_button_widget

### CODE REVIEW

File: components/basic_widgets/CheckBoxWidget.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  font-weight: normal;
+  font-style: normal;
+  text-decoration: none;
+  font-size: 15px;
+  color: rgb(183, 183, 183);
+  visibility: visible;
+  border: 2px solid rgb(17, 17, 17);
+  border-radius: 0px;
+  box-shadow: none;
+  z-index: 3;
```
Comment (eli88fine) [inline components/basic_widgets/CheckBoxWidget.vue:60]: can this z-index be removed? it doesn't look like it's having an effect

File: tests/e2e/basic_widgets/checkbox-widget.e2e.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    screenshot_path_base,
+    "basic-checkbox-empty-text"
+  );
+  await testcafe_page_visual_regression(t, screenshot_path);
+});
+
+fixture`basic_widgets/checkbox-widget/x-y-offset`
+  .page // declare the fixture
+`http://localhost:8080/basic_widgets/checkbox-widget/x-y-offset`; // specify the start page
+test("testing the Radio Button Widget with a x y offset Entered", async (t) => {
```
Comment (eli88fine) [inline tests/e2e/basic_widgets/checkbox-widget.e2e.js:59]: typo: text says "Radio Button"

**Review by eli88fine — COMMENTED** [review]
looks good.  just two small notes/questions.  also, it looks like this branch needs to be rebased to be up-to-date with development

File: components/basic_widgets/CheckBoxWidget.vue (hunk 1 above)
Comment (robinr) [reply components/basic_widgets/CheckBoxWidget.vue:60]: Agreed, I will remove.

File: tests/e2e/basic_widgets/checkbox-widget.e2e.js (hunk 2 above)
Comment (robinr) [reply tests/e2e/basic_widgets/checkbox-widget.e2e.js:59]: Corrected.

**Review by eli88fine — APPROVED** [review]

**Pull request merged**